### PR TITLE
Remove GSA functionality

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
Create separate GlobalSensitivityAnalysis.jl package and move GSA functionality to there

### CODE REVIEW

File: src/DiffEqSensitivity.jl [hunk 1]
```diff
@@ -2,42 +2,35 @@ module DiffEqSensitivity
 
 using DiffEqBase, ForwardDiff, Tracker, FiniteDiff, Statistics
 using DiffEqCallbacks, QuadGK, RecursiveArrayTools, LinearAlgebra
-using RecursiveArrayTools, QuasiMonteCarlo, Adapt
-using Parameters: @unpack, @with_kw
-using FFTW, Distributions, Requires
+using RecursiveArrayTools, Adapt
```
Comment (Vaibhavdixit02) [inline src/DiffEqSensitivity.jl:5]:
```suggestion
using Adapt
```

**Comment by ChrisRackauckas** [comment]
This is very breaking. It should first re-export GlobalSensitivity.jl (which would require tagging first) and add a warning on the appropriate functions to lead the users to GlobalSensitivity.jl.

**Comment by ChrisRackauckas** [comment]
@Vaibhavdixit02 bump on the reexporting with a depwarn?

**Comment by ChrisRackauckas** [comment]
🎉

**Pull request merged**